graphql/resolvers/exerciseCrud.ts

```diff
@@ -91,3 +92,67 @@ export const deleteExercise = async (
     }
   })
 }
+
+export const flagExercise = async (
+  _parent: void,
+  arg: MutationFlagExerciseArgs,
+  { req }: Context
+): Promise<Exercise> => {
+  const { id } = arg
+
+  const authorId = req.user?.id
+  if (!authorId) throw new Error('No User')
+
+  const exercise = await prisma.exercise.findUnique({
+    where: {
+      id
+    }
+  })
+
+  if (!isAdmin(req) && exercise?.authorId !== authorId) {
```

Comment:
anyone can flag an exercise. 

If I am a user and I come across an offensive exercise, I would flag it as inappropriate content.

I am not an admin, and I am not the author of the exercise